### PR TITLE
Migrate aabb nodes and add helper script for generating xml file

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
Migrate aabb nodes and add the helper script for generating xml file
the script can be the starting point of customer friendly new world

Demo of how script works

https://user-images.githubusercontent.com/5900509/162545045-664d4129-e164-49ac-9b38-fd3f3273e657.mp4

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake [hunk 1]
```diff
@@ -127,6 +127,7 @@ set(FILES
     Include/ScriptCanvas/Libraries/Logic/TargetedSequencer.cpp
     Include/ScriptCanvas/Libraries/Logic/WeightedRandomSequencer.cpp
     Include/ScriptCanvas/Libraries/Logic/While.cpp
+    Include/ScriptCanvas/Libraries/Math/AABBFunctions.cpp
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake:130]: I'm leaning towards just calling it AABB.h/cpp

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py:10]: This need a documentation block that explains what it's for. I'm not sure we'll ship it, but I wouldn't want it to ship without a comment that explains it and a - - help usage command

**Review by lsemp3d — APPROVED** [review]
Just a couple small comments, looks good

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py (hunk 2 above)
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py:10]: yeah, I don't want to release it until it is mature
need a better place to put it for internal use and revise, probably I can create a separate branch for it.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        sub_element.set('Name', function)
+
+def prettify_xml_content(root: ElementTree.Element) -> bytes:
+    rough_string: str = ElementTree.tostring(root, 'utf-8')
+    reparsed = minidom.parseString(rough_string)
+    return reparsed.toprettyxml(indent="    ", encoding='utf-8')
+
+if __name__ == '__main__':
+    try:
+        collect_arguments_info()
```
Comment (carlitosan) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py:142]: Do you want the extra white space here or does python demand it?

File: Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake (hunk 1 above)
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake]: done

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py (hunk 3 above)
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/generate_scriptcanvas_function_xml.py]: it is not required to do so, but I can add extra line for clear separation.
In any way, I have moved the script out of PR

**Review by carlitosan — APPROVED** [review]

**Pull request merged**